### ISSUE TITLE
Training results fluctuate significantly

### ISSUE THREAD

**Issue by Z1740220020** [ZrrSkywalker/MonoDETR#18]
Thank you so much for doing such a great job and providing great code. I'm having problems reproducing your results. I used 4 3090 graphics cards, and the settings of parameters such as batchsize and learning rate are all set by default. I find that AP always fluctuates between 17 and 19. The fluctuations are large. Is this normal?

**Comment by SibylGao** [comment]
Same question, but i got more higher score in 3D AP R40: 27.8836, 19.3681, 15.9685(test set) with default parameters.

**Comment by pummi823** [comment]
Same question . I run many times but the highest score is around19.0

**Comment by Z1740220020** [comment]
> Same question, but i got more higher score in 3D AP R40: 27.8836, 19.3681, 15.9685(test set) with default parameters.

May I ask what kind of hardware equipment you train on? I was wondering if it could be due to the fluctuations caused by multi-GPU training. The author mentioned in the paper that he only used a single 3090. And I used four 3090 graphics cards and the batch size and learning rate were not adjusted (the same as in the paper).

**Comment by Z1740220020** [comment]
> Same question . I run many times but the highest score is around19.0

May I ask how many graphics cards you used for training?

**Comment by pummi823** [comment]
> > Same question . I run many times but the highest score is around19.0
> 
> May I ask how many graphics cards you used for training?

Just one

**Comment by SibylGao** [comment]
> > Same question, but i got more higher score in 3D AP R40: 27.8836, 19.3681, 15.9685(test set) with default parameters.
> 
> May I ask what kind of hardware equipment you train on? I was wondering if it could be due to the fluctuations caused by multi-GPU training. The author mentioned in the paper that he only used a single 3090. And I used four 3090 graphics cards and the batch size and learning rate were not adjusted (the same as in the paper).

I use 4 v100 graphics cards.

**Comment by Shinkair** [comment]
> Thank you so much for doing such a great job and providing great code. I'm having problems reproducing your results. I used 4 3090 graphics cards, and the settings of parameters such as batchsize and learning rate are all set by default. I find that AP always fluctuates between 17 and 19. The fluctuations are large. Is this normal?

Hi, I am wondering whether the batchsize will influence the results, i made it as 4 (not 16), but got much worse AP. Did you guys continue trying to rise up the AP

**Comment by tranquan687** [comment]
> > Thank you so much for doing such a great job and providing great code. I'm having problems reproducing your results. I used 4 3090 graphics cards, and the settings of parameters such as batchsize and learning rate are all set by default. I find that AP always fluctuates between 17 and 19. The fluctuations are large. Is this normal?
> 
> Hi, I am wondering whether the batchsize will influence the results, i made it as 4 (not 16), but got much worse AP. Did you guys continue trying to rise up the AP

have you solved this issue? I have the same problem.

**Comment by Hyeonjeong37** [comment]
Can I ask you how to run this code with multiple gpus?
